clang/lib/CIR/CodeGen/CIRGenOpenACCClause.cpp

```diff
@@ -355,6 +356,110 @@ class OpenACCClauseCIREmitter final
     }
   }
 
+  template <typename RecipeTy>
+  RecipeTy getOrCreateRecipe(ASTContext &astCtx, const Expr *varRef,
+                             DeclContext *dc, QualType baseType,
+                             mlir::Value mainOp) {
+    mlir::ModuleOp mod =
+        builder.getBlock()->getParent()->getParentOfType<mlir::ModuleOp>();
```

Comment:
```suggestion
        builder.getBlock()->getParentOfType<mlir::ModuleOp>();
```
`getParentOfType` walks the hierarchy until it finds the requested type.

Reply:
`mlir::Block` doesn't seem to have `getParentOfType`?  I think that must be why this uses `getParent`.  OR am I missing something further?